### PR TITLE
Nef integration

### PR DESCRIPTION
**Pull request by truizlop**
## Goal

- Automate documentation verification and deployment.
- Add initial pages for documentation.

## Implementation details

nef has been used to compile docs written in Swift Playground format. A Swift project has been added to `contents/Documentation` with initial playgrounds for each page. Also, a main page has been added. Contents for this pages has been extracted from the README or the Bow release article.

The landing page for the site has been modified to include a link to this docs, as can be seen in this screenshot:

Landing page

#### Travis

  * For the Travis configuration to work, Homebrew installable dependencies have been set as part of the gobal configuration, which we've found to be the best approach regarding this.

### CODE REVIEW

File: contents/Documentation/Documentation/Info.plist [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	<key>CFBundleName</key>
+	<string>$(PRODUCT_NAME)</string>
+	<key>CFBundlePackageType</key>
+	<string>FMWK</string>
+	<key>CFBundleShortVersionString</key>
+	<string>1.0</string>
+	<key>CFBundleVersion</key>
+	<string>1.0</string>
+	<key>NSHumanReadableCopyright</key>
+	<string>Copyright © 2019. The NEF authors.</string>
```
Comment (miguelangel-dev) [inline contents/Documentation/Documentation/Info.plist]:
```suggestion
	<string>Copyright © 2019. The nef authors.</string>
```

Comment (miguelangel-dev) [reply contents/Documentation/Documentation/Info.plist]: I have changed it [in this PR](https://github.com/bow-swift/nef/pull/23)

File: contents/Documentation/Podfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+target 'Documentation' do
+  platform :osx, '10.14'
+  use_frameworks!
+
+  pod "Bow", "~> 0.3.0"
```
Comment (miguelangel-dev) [inline contents/Documentation/Podfile]: maybe we could delete `, "~> 0.3.0"` in order to keep it updates - the user will know how to fix the version

Comment (miguelangel-dev) [inline contents/Documentation/Podfile]: maybe for documentation, we should delete `, "~> 0.3.0"`to keep it updates to the last version - we should review how to compile using a determined branch/version as we did in playground generation.

File: contents/Documentation/Podfile.lock [hunk 3]
```diff
@@ -0,0 +1,81 @@
+PODS:
```
Comment (miguelangel-dev) [inline contents/Documentation/Podfile.lock]: I think we could add this file to `.gitignore` for documentation

File: contents/Documentation/Config/Info.plist [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+	<key>CFBundleName</key>
+	<string>$(PRODUCT_NAME)</string>
+	<key>CFBundlePackageType</key>
+	<string>FMWK</string>
+	<key>CFBundleShortVersionString</key>
+	<string>1.0</string>
+	<key>CFBundleVersion</key>
+	<string>1.0</string>
+	<key>NSHumanReadableCopyright</key>
+	<string>Copyright © 2019. The NEF authors.</string>
```
Comment (miguelangel-dev) [inline contents/Documentation/Config/Info.plist]:
```suggestion
	<string>Copyright © 2019. The nef authors.</string>
```

File: contents/Documentation/Podfile (hunk 2 above)
Comment (truizlop) [reply contents/Documentation/Podfile]: Exactly. Right now we can resort to the latest published version, but we will need to adjust accordingly to compile against the last commit on the branch that is been pushed.

**Comment by calvellido** [comment]
Have you guys checked that `nef` can be installed that way in the Travis image?  I'm not able to generate the docs through `nef` in a forked project.

**Comment by calvellido** [comment]
Full log of the failed job:

https://api.travis-ci.org/v3/job/511934230/log.txt

What I'm finding weird is that while `homebrew` seems to be around and being triggered, it is only installing `sourcekitten`, but I'm not finding any reference to `nef` in the logs.

**Comment by miguelangel-dev** [comment]
> Full log of the failed job:
> 
> https://api.travis-ci.org/v3/job/511934230/log.txt
> 
> What I'm finding weird is that while `homebrew` seems to be around and being triggered, it is only installing `sourcekitten`, but I'm not finding any reference to `nef` in the logs.

Yep, it is not installed - `/Users/travis/.travis/functions: line 104: nef: command not found` - do you have travis configuration updated with the `hombrew command`
```
sudo: required
addons:
  homebrew:
    taps:
      - bow-swift/nef
    packages:
      - nef
```
maybe for any reason is it different or not work properly in images?

**Comment by truizlop** [comment]
@calvellido it is very strange, you can see in the jobs for this repo that it is actually installed and running. Can you post your fork?

**Comment by calvellido** [comment]
Sure, the fork is just this branch already merged:

https://github.com/calvellido/bow

**Comment by calvellido** [comment]
I don't think any action in this repo has already triggered the `deploy-docs` job, which is the one actually running `nef`

**Comment by calvellido** [comment]
After the `nef` addition I mean, of course.

**Pull request merged**